src/components/instant-apps-interactive-legend/support/helpers.ts

```diff
@@ -145,8 +146,9 @@ export function getCategoryData(
   return parentLegendElementInfoData ? parentLegendElementInfoData?.nestedInfos?.[infoIndex] : data?.categories?.get(elementInfo?.label ?? layer?.id);
 }
 
-export function getIntLegendLayerData(fLayer: __esri.FeatureLayer, data): IIntLegendLayerData {
-  return data?.[fLayer?.id];
+export function getIntLegendLayerData(fLayer: __esri.FeatureLayer): IIntLegendLayerData {
+  const data = interactiveLegendState?.data?.[fLayer?.id];
```

Comment:
This could be returned instead of creating a new variable and not doing anything with it. Also, the return type for this would have better checking if it was `IIntLegendLayerData | undefined` since that's what's actually being returned due to the optional chaining.

Reply:
Done. 

Keeping the return type the same for now, changing it will affect other functions. Will address in the future when we're done patching.